daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -490,8 +503,22 @@ codegenTests codegenDir = testGroup "daml codegen"
                     callCommandQuiet $ unwords ["daml new", projectDir, "skeleton"]
                     withCurrentDirectory projectDir $ do
                         callCommandQuiet "daml build"
-                        let darFile = projectDir</> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
-                            outDir  = projectDir</> "generated" </> lang
+                        let darFile = projectDir </> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
+                            outDir  = projectDir </> "generated" </> lang
+                        when (lang == "ts") $ do
+                          createDirectoryIfMissing True "generated"
+                          withCurrentDirectory "generated" $ do
+                            -- SDK version is 0.0.0; daml2ts needs
+                            -- 'daml-types' to be here in the filesystem...
+                            copyDirectory damlTypes "daml-types"
+                            BSL.writeFile "package.json" $ encode (
+                            -- ... and this package.json so it can find it.
+                              object
+                                [ "private" .= True
+                                , "workspaces" .= [T.pack lang]
+                                , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
+                                ]
+                              )
```

Comment:
Why not spend more 💵s?
```suggestion
                            BSL.writeFile "package.json" $ encode $
                            -- ... and this package.json so it can find it.
                              object
                                [ "private" .= True
                                , "workspaces" .= [T.pack lang]
                                , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
                                ]
```

Reply:
Lol :) Ok